types/src/v0/impls/header.rs

```diff
@@ -1099,6 +1100,30 @@ impl BlockHeader<SeqTypes> for Header {
             )?,
         })
     }
+
+    fn auth_root(&self) -> anyhow::Result<Option<[u8; 32]>> {
+        match self {
+            Header::V1(_) | Header::V2(_) | Header::V3(_) => Ok(None),
+            Header::V4(header) => {
+                let mut reward_root_bytes = Vec::new();
+                let el_root = [0; 32];
+                let placeholder_root = [0; 32];
+                header
+                    .reward_merkle_tree_root
+                    .serialize_compressed(&mut reward_root_bytes)?;
```

Comment:
q: why do we need to serialize again? we should just take the root value (which is already a `[u8; 32]`, correct? 

we don't want to do any extra serialization since this step will have to be done/reproduced in contract as well during verification. so we want to avoid any solidity-unfriendly computation.

Reply:
@mrain do you know? I can't recall on top of my head.

Reply:
If we are pointing to `jf-merkle-tree` 0.1, the `MerkleTreeCommitment` contains more than 32 bytes: https://github.com/EspressoSystems/jellyfish/blob/0.4.5/merkle_tree/src/internal.rs#L86-L108

Reply:
i see, two things to keep in mind.
1. we should keep this `reward_root_bytes` fixed size, (doesn't have to be 32 bytes, but fixed for each version)
2. anything calculation we do in this function, we will have to do in contract, thus all design needs solidity friendliness

I'd recommend `reward_root_bytes := [ root | height | size]` which takes 32 + 8 + 8 = 48 bytes exactly. 

in the contract, we would just accept `bytes reward_root_bytes` but first thing is to check if `48` in length.

Reply:
I think in the rust code we can add a type for the rewards MT auth root commitment to make it clear that this it different from the original jellyfish rewards MT Commitment.

Reply:
In the current impl, it's unsafe to ignore the `height` because it opens the extension attack.
Either double-hashing or committing the MT height will protect us.

Reply:
More clear version:
- If we commit the height, e.g. using `comm = keccak(digest, height, ...)` we should reject all proofs with different lengths.
- In our case, with double-hashing, we can rely on digest w/o height, num_leaves. 

cc @alxiong

Reply:
cc @imabdulbasit we can use the digest directly for the auth root input.